web/frontend/src/components/slam/index.svelte

```diff
@@ -85,32 +117,38 @@ const refresh2d = async () => {
       lastTimestamp = mapTimestamp;
     }
   } catch (error) {
-    refreshErrorMessage2d = error !== null && typeof error === 'object' && 'message' in error
-      ? `${refreshErrorMessage} ${error.message}`
-      : `${refreshErrorMessage} ${error}`;
+    refreshErrorMessage2d =
+      error !== null && typeof error === 'object' && 'message' in error
+        ? `${refreshErrorMessage} ${error.message}`
+        : `${refreshErrorMessage} ${error}`;
   }
 };
 
 const refresh3d = async () => {
   try {
-    const mapTimestamp = await getLatestMapInfo($robotClient, name);
-
-    /*
-     * The map timestamp is compared to the last timestamp
-     * to see if a change has been made to the pointcloud map.
-     * A new call to getPointCloudMap is made if an update has occured.
-     */
-    if (mapTimestamp?.getSeconds() !== lastTimestamp.getSeconds()) {
-      pointcloud = await getPointCloudMap($robotClient, name);
-    }
-    if (mapTimestamp) {
-      lastTimestamp = mapTimestamp;
+    if (overrides?.isMapping && overrides?.getMappingSessionPCD) {
```

Comment:
Do you anticipate any issues with overriding instead in the `api` folder? I think that would probably scale better: if in the future some other developer for some reason uses this API in another component they won't accidentally foot-gun themselves because the override exists at the component layer when it should be at the api layer.

Reply:
🦶🔫

So I was wondering about this too. Would your expectation be that the override behavior is available wherever you use the API and that you should be able to override it's functionality anywhere you use those APIs? To me it does feel like the intention is to overwrite the behavior of this particular component rather than the behavior of the underlying APIs it's using, allowing its rendering and functionality to change depending on where the RC cards are being rendered. Thoughts?

Reply:
Yeah that's fair, and when I think about the API folder more it's going to be something that gets abstracted away in the TS SDK eventually. All's good! Thanks for letting me think out loud 😂